lndmanage/lib/liquidityhints.py

```diff
@@ -17,12 +21,22 @@
 TIME_NODE_IS_SLOW_SEC = 5  # the time a node is viewed as slow
 
 
-class ShortChannelID(int):
-    pass
+@dataclass
+class AmountHistory:
+    amount: int = 0  # msat
+    timestamp: int = 0
+
+    def __bool__(self):
+        return bool(self.amount) and bool(self.timestamp)
+
+    def __gt__(self, other):
+        return self and self.amount > other.amount
```

Comment:
I guess this should check  for truthiness of `other` as well?
Like `return self and (not other or self.amount > other.amount)`

Also it looks like since `bool(x) == False` if `x.amount == 0`, for any `x: x.amount == 0` and any `y`, `x < y && x > y`.
So this would break the consistency assumptions for these operators.

Maybe not a practical issue here, just bringing it up for consideration.

Reply:
good comment, I changed it to also check for bool(other), it also simplifies the comparison, before I always checked truthiness explicitly